packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol

```diff
@@ -94,7 +94,7 @@ contract OptimisticAsserter is OptimisticAsserterInterface, Lockable, Ownable {
         uint256 bond,
         uint256 liveness,
         bytes32 identifier
-    ) public returns (bytes32) {
+    ) public nonReentrant() returns (bytes32) {
         asserter = asserter == address(0) ? msg.sender : asserter;
         bytes32 assertionId = _getId(claim, bond, liveness, currency, callbackRecipient, sovereignSecurity, identifier);
 
```

Comment:
unrelated to the code in this PR but you should also pull the `currency.safeTransferFrom` logic to the bottom to follow the check effects interaction pattern

Reply:
Agree, moved token transfers after state update in 65dcb9ef33a03b1b76d83bce1623ed579b246155